packages/store/src/usersSlice/thunks.ts

```diff
@@ -377,33 +462,42 @@ const customPermissionHandler = (accu, permission) => {
   };
 };
 
-const mapPermissionSet = (permissionSetName, names, scope, existingGroupsPermissions = {}) => {
-  const permission = Object.values(uiPermissionsById).find(permission => permission.permissionSets[scope] === permissionSetName).value;
+const mapPermissionSet = (
+  permissionSetName: PermissionSetIds,
+  names: string[],
+  scope: UiPermissionsByAreaKeys,
+  existingGroupsPermissions: CombinedPermission = {}
+) => {
+  const permission = Object.values(uiPermissionsById).find(permission => permission.permissionSets[scope] === permissionSetName)?.value as UiPermissionsByIdKey;
   const scopedPermissions = names.reduce((accu, name) => combinePermissions(accu, { [name]: [permission] }), existingGroupsPermissions);
   return Object.entries(scopedPermissions).reduce((accu, [key, permissions]) => ({ ...accu, [key]: deriveImpliedAreaPermissions(scope, permissions) }), {});
 };
 
-const isEmptyPermissionSet = permissionSet =>
+const isEmptyPermissionSet = (permissionSet: Partial<UiPermissions>) =>
   !Object.values(permissionSet).reduce((accu, permissions) => {
     if (Array.isArray(permissions)) {
       return accu || !!permissions.length;
     }
     return accu || !isEmpty(permissions);
   }, false);
 
-const parseRolePermissions = ({ permission_sets_with_scope = [], permissions = [] }, permissionSets) => {
+//TODO: remove `permissions` if that's legacy
```

Comment:
We won't migrate the old roles definitions (+ never removed the deprecated endpoints as they were actively used), so can't remove them here.